p2p/src/test/kotlin/maru/p2p/P2PTest.kt

```diff
@@ -894,12 +889,23 @@ class P2PTest {
         assertNetworkIsConnectedToPeer(p2pNetworkImpl2, PEER_ID_NODE_1)
       }
 
+      val peer1 = p2pNetworkImpl1.maruPeerManager.getPeers()[0]
+      val peer2 = p2pNetworkImpl2.maruPeerManager.getPeers()[0]
+
+      val peer1List = mutableListOf<ULong>()
+      val peer2List = mutableListOf<ULong>()
       // check for the next 10 seconds that the peers are still connected
       val startTime = System.currentTimeMillis()
-      while (System.currentTimeMillis() < startTime + 100000L) {
+      while (System.currentTimeMillis() < startTime + 11000L) {
         assertNetworkIsConnectedToPeer(p2pNetworkImpl1, PEER_ID_NODE_2)
         assertNetworkIsConnectedToPeer(p2pNetworkImpl2, PEER_ID_NODE_1)
+        peer1List.add(peer1.getStatus()!!.latestBlockNumber)
+        peer2List.add(peer2.getStatus()!!.latestBlockNumber)
+        sleep(500)
       }
+
+      assertThat(peer1List).contains(0uL, 1uL, 2uL, 3uL, 4uL, 5uL, 6uL, 7uL, 8uL, 9uL)
```

Comment:
It would be nice to make this logic more cohesive by reusing the exact same list that is used for mocking and tying its size to the `11000L` above that is used in a loop end condition